build.rs

```diff
@@ -33,20 +34,30 @@ fn main() {
     let mut source_code = String::from(
         "use std::collections::HashMap;
     
-pub fn template_files() -> HashMap<&'static str, Vec<(&'static str, &'static str)>> {
+pub fn template_files(
+) -> HashMap<&'static str, HashMap<&'static str, Vec<(&'static str, &'static str)>>> {
```

Comment:
This type is a bit hard to decipher, maybe we could use structs or type defs to make the semantics clearer?

Reply:
Do you know of a way to shared types between the `build.rs` and the rest of the code? If I can do that, it'll also help me solve [your other comment](https://github.com/amethyst/tools/pull/88#discussion_r295299392)

Reply:
The only way to do that would be to put the type into an extra crate and use that as both a build dependency and a real dependency for the project.